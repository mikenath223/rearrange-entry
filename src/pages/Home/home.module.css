.wrapper {
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 100vh;
  width: min(500px, 99%);
  margin: 100px auto 50px auto;
}

.divider {
  width: 250px;
  margin: 20px 0;
  color: grey;
}