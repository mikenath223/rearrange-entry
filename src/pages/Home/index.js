import { useState } from 'react';
import { List, Input, Button } from 'components';
import { Grid, Typography } from '@material-ui/core';
import { v4 as uuidv4 } from 'uuid';
import styles from './home.module.css';

const HomePage = () => {
  const [entry, setEntry] = useState('_');
  const [listItems, setListItems] = useState([]);

  const handleAddEntry = () => {
    const item = { id: uuidv4(), entry };

    setListItems([item, ...listItems]);
    setEntry('');
  };

  return (
    <main className={styles.wrapper}>
      <Typography variant="h5" gutterBottom>
        Reorder Phone Numbers
      </Typography>
      <Grid container direction="column" spacing={3}>
        <Grid item xs={7}>
          <Input
            entry={entry}
            handleSetEntry={val => setEntry(val)}
          />
        </Grid>
        <Grid item xs={4}>
          <Button handleClick={handleAddEntry} isDisabled={entry.length !== 10} />
        </Grid>
      </Grid>
      <hr className={styles.divider} />
      <Typography variant="h6" gutterBottom>
        Phone Numbers
      </Typography>
      <List listItems={listItems} handleSetListItems={setListItems} />
    </main>
  );
};

export default HomePage;
