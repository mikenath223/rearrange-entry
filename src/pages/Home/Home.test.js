import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Home from './index';

describe('Home', () => {
  it('should match snapshot', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  it('should add phone number to list', async () => {
    const { getByRole } = render(<Home />);
    const input = getByRole('input');
    const button = getByRole('button');

    const testNumber = '(123) 456-7890';
    userEvent.type(input, testNumber);
    userEvent.click(button);
    const listItem = await screen.findByRole('listitem');

    expect(listItem.textContent).toBe(testNumber);
  });
});
