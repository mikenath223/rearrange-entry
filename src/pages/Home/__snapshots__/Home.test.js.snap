// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <main
    class="wrapper"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Reorder Phone Numbers
    </h5>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
          >
            Enter phone number
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              inputmode="numeric"
              role="input"
              type="text"
              value="(___) ___-____"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
              >
                <span>
                  Enter phone number
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled=""
          role="button"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </span>
            Add
          </span>
        </button>
      </div>
    </div>
    <hr
      class="divider"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    >
      Phone Numbers
    </h6>
    <div
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="droppable"
      style="background: lightgrey; padding: 8px; width: 250px; min-height: 50px;"
    />
  </main>
</div>
`;
